### PR TITLE
test(bigquery): Add tests to verify lossless max timestamp parsing

### PR DESCRIPTION
**Pull request by jinseopkim0**
b/326260769

### CODE REVIEW

File: java-bigquery/google-cloud-bigquery/src/test/java/com/google/cloud/bigquery/it/ITBigQueryTest.java [hunk 1]
```diff
@@ -1215,6 +1215,28 @@ static GoogleCredentials loadCredentials(String credentialFile) {
     }
   }
 
+  @Test
+  void testLosslessMaxTimestampIntegration() throws InterruptedException {
+    // 1. Initialize BigQueryOptions with the lossless timestamp flag enabled
+    // We use RemoteBigQueryHelper to get standard test environment credentials
+    BigQueryOptions options = bigquery.getOptions().toBuilder().setUseInt64Timestamps(true).build();
```
Comment (lqiu96) [inline java-bigquery/google-cloud-bigquery/src/test/java/com/google/cloud/bigquery/it/ITBigQueryTest.java]: Can we set the int64timestamp setting via the DataFormatOptions: https://github.com/googleapis/google-cloud-java/blob/8a25165f6c08e99542ba323b43a2cbb96366deb4/java-bigquery/google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/BigQueryOptions.java#L109

File: java-bigquery/google-cloud-bigquery/src/test/java/com/google/cloud/bigquery/it/ITBigQueryTest.java [hunk 2]
```diff
@@ -1215,6 +1215,28 @@ static GoogleCredentials loadCredentials(String credentialFile) {
     }
   }
 
+  @Test
+  void testLosslessMaxTimestampIntegration() throws InterruptedException {
```
Comment (lqiu96) [inline java-bigquery/google-cloud-bigquery/src/test/java/com/google/cloud/bigquery/it/ITBigQueryTest.java:1219]: Since DataFormatOptions also has other configs, do you think we can also test for FLOAT64 and ISO8601 for the max value edge case?

File: java-bigquery/google-cloud-bigquery/src/test/java/com/google/cloud/bigquery/it/ITBigQueryTest.java (hunk 1 above)
Comment (jinseopkim0) [reply java-bigquery/google-cloud-bigquery/src/test/java/com/google/cloud/bigquery/it/ITBigQueryTest.java]: Thanks for the feedback, done.

File: java-bigquery/google-cloud-bigquery/src/test/java/com/google/cloud/bigquery/it/ITBigQueryTest.java (hunk 2 above)
Comment (jinseopkim0) [reply java-bigquery/google-cloud-bigquery/src/test/java/com/google/cloud/bigquery/it/ITBigQueryTest.java:1219]: Thanks for the feedback, done.

File: java-bigquery/google-cloud-bigquery/src/test/java/com/google/cloud/bigquery/it/ITBigQueryTest.java [hunk 3]
```diff
@@ -1215,6 +1215,56 @@ static GoogleCredentials loadCredentials(String credentialFile) {
...
+    String query = "SELECT TIMESTAMP '9999-12-31 23:59:59.999999 UTC' as max_ts";
+    QueryJobConfiguration config = QueryJobConfiguration.newBuilder(query).build();
+
+    // 1. Test lossless 64-bit integer parsing (useInt64Timestamps = true)
+    DataFormatOptions losslessOptions =
+        DataFormatOptions.newBuilder().useInt64Timestamp(true).build();
```
Comment (lqiu96) [inline java-bigquery/google-cloud-bigquery/src/test/java/com/google/cloud/bigquery/it/ITBigQueryTest.java:1225]: nit: Can you use DataFormatOptions.TimestampFormatOptions.INT64?

Comment (lqiu96) [reply java-bigquery/google-cloud-bigquery/src/test/java/com/google/cloud/bigquery/it/ITBigQueryTest.java:1225]: Oh wait, nvm. this works

**Review by lqiu96 — APPROVED** [review]

File: java-bigquery/google-cloud-bigquery/src/test/java/com/google/cloud/bigquery/it/ITBigQueryTest.java (hunk 3 above)
Comment (jinseopkim0) [reply java-bigquery/google-cloud-bigquery/src/test/java/com/google/cloud/bigquery/it/ITBigQueryTest.java:1225]: Will follow-up with a new PR to add additional tests. I would like to keep the current tests, to ensure useInt64Timestamp works.

Comment (ebyhr) [inline java-bigquery/google-cloud-bigquery/src/test/java/com/google/cloud/bigquery/it/ITBigQueryTest.java:1225]: When I tested this option using `QueryJobConfiguration.Builder#setCreateDisposition`, the data format did not use Int64Timestamp. Is this a known issue?
